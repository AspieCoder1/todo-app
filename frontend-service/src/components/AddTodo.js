import React, { Component } from 'react';
import classNames from 'classnames';

import styles from '../styles/components/addTodo.module.scss';
export default class AddTodo extends Component {
	render() {
		return (
			<form onSubmit={this.props.onSubmit}>
				<div className={styles.inputGroup}>
					<input
						className={classNames(styles.input, {
							[styles.inputError]:
								this.props.errors && this.props.errors.noTodo,
						})}
						type="text"
						name="todoToAdd"
						value={this.props.todoToAdd}
						onChange={this.props.onChange}
						placeholder="Enter your new todo here..."
					/>
					<button className={styles.button} type="submit">
						Add Todo
					</button>
				</div>
				{this.props.errors && this.props.errors.noTodo ? (
					<p className={styles.error}>Please provide a todo</p>
				) : null}
			</form>
		);
	}
}
