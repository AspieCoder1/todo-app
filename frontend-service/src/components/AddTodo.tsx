import * as React from 'react';
import classNames from 'classnames';
import * as styles from '../styles/components/addTodo.module.scss';

interface IErrors {
	[key: string]: string;
}

interface IProps {
	errors: IErrors | undefined;
	todoToAdd: string;
	onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const AddTodo: React.FC<IProps> = props => (
	<form onSubmit={props.onSubmit}>
		<div className={styles.inputGroup}>
			<input
				className={classNames(styles.input, {
					[styles.inputError]: props.errors && props.errors.noTodo,
				})}
				type="text"
				name="todoToAdd"
				value={props.todoToAdd}
				onChange={props.onChange}
				placeholder="Enter your new todo here..."
			/>
			<button className={styles.button} type="submit">
				Add Todo
			</button>
		</div>
		{props.errors && props.errors.noTodo ? (
			<p className={styles.error}>Please provide a todo</p>
		) : null}
	</form>
);

export default AddTodo;
