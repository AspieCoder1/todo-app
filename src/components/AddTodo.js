import React, { Component } from 'react';

export default class AddTodo extends Component {
	render() {
		return (
			<form onSubmit={this.props.onSubmit}>
				<input
					type="text"
					name="todoToAdd"
					value={this.props.todoToAdd}
					onChange={this.props.onChange}
				/>
				{this.props.errors && this.props.errors.noTodo ? (
					<p>Please provide a todo</p>
				) : null}
				<button type="submit">Add Todo</button>
			</form>
		);
	}
}
