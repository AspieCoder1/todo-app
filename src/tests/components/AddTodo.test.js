import React from 'react';
import { shallow } from 'enzyme';

import AddTodo from '../../components/AddTodo';

test('it should render add todo correctly', () => {
	const onSubmit = jest.fn();
	const wrapper = shallow(<AddTodo onSubmit={onSubmit} />);

	wrapper.find('input').value = 'New Todo';
	expect(wrapper).toMatchSnapshot();
});

it('should handle submit correctly', () => {
	const onSubmit = jest.fn();
	const wrapper = shallow(
		<AddTodo errors={{ noTodo: true }} onSubmit={onSubmit} />
	);

	wrapper.find('input').simulate('change', {
		target: {
			value: 'Todo Body',
		},
	});

	wrapper.update();
	wrapper.find('form').simulate('submit');
	expect(onSubmit).toHaveBeenCalledTimes(1);
});
