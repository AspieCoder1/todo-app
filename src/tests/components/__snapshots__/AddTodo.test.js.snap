// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render add todo correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="inputGroup"
  >
    <input
      className="input"
      name="todoToAdd"
      placeholder="Enter your new todo here..."
      type="text"
    />
    <button
      className="button"
      type="submit"
    >
      Add Todo
    </button>
  </div>
</form>
`;
